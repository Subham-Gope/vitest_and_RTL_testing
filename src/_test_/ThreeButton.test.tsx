// import { describe, test, expect } from "vitest";
// import { render, screen } from "@testing-library/react";
// import userEvent from "@testing-library/user-event";
// import ThreeButton from "../components/ThreeButton";

// describe("Testing the ThreeButton.tsx", () => {
//   test("selects button with tab", async () => {
//     // Setting up the userEvent
//     const user = userEvent.setup();

//     // setting up the debugger
//     screen.debug();

//     // setting up the snapshot matching
//     const { container } = render(<ThreeButton />);
//     expect(container).toMatchSnapshot();

//     const btn1 = screen.getByRole("button", { name: /first/i });
//     const btn2 = screen.getByRole("button", { name: /second/i });
//     const btn3 = screen.getByRole("button", { name: /third/i });

//     expect(btn1).toBeInTheDocument();
//     expect(btn2).toBeInTheDocument();
//     expect(btn3).toBeInTheDocument();

//     // Initiating user interaction
//     await user.tab();
//     expect(btn1).toHaveFocus();
//     await user.tab();
//     expect(btn2).toHaveFocus();
//     await user.tab();
//     expect(btn3).toHaveFocus();
//   });
// });
