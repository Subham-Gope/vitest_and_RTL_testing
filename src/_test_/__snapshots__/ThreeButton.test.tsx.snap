// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing the ThreeButton.tsx > selects button with tab 1`] = `
<div>
  <button>
    First
  </button>
  <button>
    Second
  </button>
  <button>
    Third
  </button>
</div>
`;
